### ISSUE TITLE
[css-fonts] Default angle for font-style:oblique should conform to host platform

### ISSUE THREAD

**Issue by jfkthame** [w3c/csswg-drafts#2295]
Currently the Fonts 4 draft says that if the optional `<angle>` is not specified for `font-style: oblique`, this "represents an angle of "20deg"". However, in my experience the angle used for synthetic-oblique font styling -- which has long been provided by most platform text-drawing APIs, when asked to draw "Italic" text with a font family that only provides Regular -- varies significantly between platforms, with (for example) Windows applying a more pronounced slant than macOS.

In order to better match the "native" text rendering of the platform, therefore, Firefox currently uses a platform-specific value when applying a synthetic-italic/oblique effect. I would suggest that this is appropriate behavior for `font-style: oblique` when no explicit `<angle>` is given, and rather than mandating a specific angle in this case, the spec should say something along the lines of "if an `<angle>` is not specified, the user agent should use a slant angle that harmonizes with the platform's typical text rendering behavior."

Moreover, if we *must* specify a default here, rather than allowing the browser to follow platform conventions, I think `20deg` is too large. Current Firefox behavior -- which was designed to harmonize with host platform behavior -- uses an angle between `11.3deg` and `16.7deg`. Checking the `italicAngle` field in a sampling of actual italic/oblique fonts shows values that cluster around 11°-12°, with occasional outliers as small as 6° or as large as 16.3°.

[1] https://drafts.csswg.org/css-fonts-4/#valdef-font-style-oblique-angle

**Comment by svgeesus** [comment]
>  Checking the italicAngle field in a sampling of actual italic/oblique fonts shows values that cluster around 11°-12°, with occasional outliers as small as 6° or as large as 16.3°.

Interesting data, which itself indicates that 20deg is too large.

> Current Firefox behavior -- which was designed to harmonize with host platform behavior -- uses an angle between 11.3deg and 16.7deg

Could you clarify which angle is used on which platform?

In general I dislike platform-dependent or UA-dependent behaviour, but in this case there is no particular downside to interop (stylesheet authors can always select the value they want, this handling of unspecified value is just for backwards compat). It does makes default-angle testing a bit harder. But I would not oppose this if that is what we decide.

**Comment by litherum** [comment]
Oh, I picked 20deg because I rendered some characters in a couple non-variable fonts which I thought were representative, took a screenshot, and used a compass and rulers to figure out what their angle was. It’s totally possible that either

1. The rendered angle doesn’t match the information in the font file
2. My fonts were not representative

I’m all for correcting the measurements I’ve made.

Making the threshold be platform-specific is problematic, because it can cause a totally different font to be chosen for the same input. For example, if there are fonts on the site at angles 19deg and 45deg, moving the threshold from 20deg to 18deg will cause an apparent visual shift of 45deg-19deg=26deg. This is a huge visual difference between platforms with only a tiny difference in threshold.
